hail/src/main/scala/is/hail/annotations/Region.scala

```diff
@@ -156,6 +156,32 @@ object Region {
   def setMemory(offset: Code[Long], size: Code[Long], b: Code[Byte]): Code[Unit] =
     Code.invokeScalaObject[Long, Long, Byte, Unit](Region.getClass, "setMemory", offset, size, b)
 
+  def firstNonZeroByteOffset(address: Code[Long], nBits: Code[Long]): Code[Long] =
+    Code.invokeScalaObject[Long, Long, Long](Region.getClass, "firstNonZeroByteOffset", address, nBits)
+
+  def firstNonZeroByteOffset(address: Long, nBits: Long): Long = {
+    val m1 = nBits / 32L
+    var i = 0L
+
+    while(i < m1) {
+      if(loadInt(address + i * 4) != 0) {
+        return i
+      }
+      i += 1
+    }
+
+    i = i * 32L
+    while(i < nBits) {
```

Comment:
you should totally add another loop in between that loads bytes. It's common to have arrays with tens of elements where we won't benefit from the 32-bit vectorization.